File: frontend/src/types/case.ts
export enum DifficultyLevel {
  BASIC = 'basic',
  INTERMEDIATE = 'intermediate',
  ADVANCED = 'advanced',
}

export interface EvidenceItem {
  number: string;
  description: string;
}

export interface CaseAttachment {
  id: string;
  file_name: string;
  file_type: string;
  file_size: number;
  uploaded_at: string;
}

export interface Case {
  id: string;
  title: string;
  description: string;
  background_facts: string;
  relevant_laws: string;
  difficulty_level: DifficultyLevel;
  evidence_items: EvidenceItem[];
  created_by: string;
  created_at: string;
  updated_at: string;
  attachments: CaseAttachment[];
}

export interface CaseListItem {
  id: string;
  title: string;
  difficulty_level: DifficultyLevel;
  created_by: string;
  created_at: string;
}

File: backend/app/models/case.py
import uuid
from datetime import datetime, timezone
from typing import Optional
from sqlalchemy import String, Text, ForeignKey, Integer, Enum as SAEnum
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy.dialects.postgresql import UUID, JSONB

from app.database import Base
from app.core.enums import DifficultyLevel


class Case(Base):
    __tablename__ = "cases"

    id: Mapped[str] = mapped_column(UUID(as_uuid=False), primary_key=True, default=lambda: str(uuid.uuid4()))
    title: Mapped[str] = mapped_column(String(200), nullable=False)
    description: Mapped[str] = mapped_column(Text, nullable=False)
    background_facts: Mapped[str] = mapped_column(Text, nullable=False)
    relevant_laws: Mapped[str] = mapped_column(Text, nullable=False)
    difficulty_level: Mapped[str] = mapped_column(
        SAEnum(DifficultyLevel, name="difficulty_level"), default=DifficultyLevel.INTERMEDIATE
    )
    # Evidence items: list of {"number": "证据1", "description": "合同原件..."}
    evidence_items: Mapped[Optional[list]] = mapped_column(JSONB, nullable=True, default=list)
    created_by: Mapped[str] = mapped_column(UUID(as_uuid=False), ForeignKey("users.id"), nullable=False)
    created_at: Mapped[datetime] = mapped_column(default=lambda: datetime.now(timezone.utc))
    updated_at: Mapped[datetime] = mapped_column(
        default=lambda: datetime.now(timezone.utc),
        onupdate=lambda: datetime.now(timezone.utc),
    )

    attachments: Mapped[list["CaseAttachment"]] = relationship(back_populates="case", cascade="all, delete-orphan")


class CaseAttachment(Base):
    __tablename__ = "case_attachments"

    id: Mapped[str] = mapped_column(UUID(as_uuid=False), primary_key=True, default=lambda: str(uuid.uuid4()))
    case_id: Mapped[str] = mapped_column(UUID(as_uuid=False), ForeignKey("cases.id", ondelete="CASCADE"), nullable=False)
    file_name: Mapped[str] = mapped_column(String(255), nullable=False)
    file_path: Mapped[str] = mapped_column(String(500), nullable=False)
    file_type: Mapped[str] = mapped_column(String(50), nullable=False)
    file_size: Mapped[int] = mapped_column(Integer, nullable=False)
    uploaded_at: Mapped[datetime] = mapped_column(default=lambda: datetime.now(timezone.utc))

    case: Mapped["Case"] = relationship(back_populates="attachments")
